specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1296 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+                     | calling `sbi_tee_host_run_tvm_vcpu()`.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_create_tvm_errors]
+.TEE Host Create TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `tvm_create_params_addr` was invalid.
+| SBI_ERR_INVALID_PARAM   | `tvm_create_params_len` was invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Host Finalize TVM (FID #6)
+[source, C]
+------
+struct sbiret sbi_tee_host_finalize_tvm(unsigned long tvm_guest_id);
+------
+Transitions the TVM specified by `tvm_guest_id` from the "TVM_INITIALIZING" state to a "TVM_RUNNABLE"
+state. The host must finalize TVM shared-memory execution parameters like the entry point (`ENTRY_PC`)
+and boot argument (`ENTRY_ARG`) on the boot VCPU configured by `sbi_tee_host_create_tvm()` before making
+this call. 
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_finalize_tvm_errors]
+.TEE Host Finalize TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM | `tvm_guest_id` was invalid, or the
+                          TVM wasn't in the `TVM_INITIALIZING` state.
+| SBI_ERR_FAILED        | The operation failed for unknown reasons.
+|===
+ 
+=== Function: TEE Host Destroy TVM (FID #7)
+[source, C]
+-------
+struct sbiret sbi_tee_host_destroy_tvm(unsigned long tvm_guest_id);
+-------
+
+Destroys a confidential TVM previously created using *`sbi_tee_host_create_tvm()`*.  
+
+Confidential TVM memory is automatically released following successful destruction, and it
+can be assigned to other TVMs. Repurposing confidential memory for use by non-confidential
+TVMs requires an explicit call to *`sbi_tee_reclaim_pages()`* (described below).
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_destroy_tvm_errors]
+.TEE Host Destroy TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM | `tvm_guest_id` was invalid.
+| SBI_ERR_FAILED        | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Host Add TVM Memory Region (FID #8)
+[source, C]
+-----
+struct sbiret sbi_tee_host_add_tvm_memory_region(unsigned long tvm_guest_id,
+                                                 unsigned long tvm_memory_region_type,
+                                                 unsigned long tvm_gpa_addr,
+                                                 unsigned long region_len);
+-----
```

Comment:
`tvm_add_memory_region` has changed to only allow confidential memory regions, so the `tvm_memory_region_type` has been dropped. The text below needs to be updated as well.